openassessment/workflow/team_api.py

```diff
@@ -0,0 +1,139 @@
+"""
+Functions that largely mirror the workflow functions specified
+in `workflow.api`, but specifically for handling team submissions.
+"""
+import logging
+
+from openassessment.workflow.errors import (
+    AssessmentWorkflowInternalError,
+    AssessmentWorkflowRequestError,
+    AssessmentWorkflowNotFoundError
+)
+from openassessment.workflow.models import TeamAssessmentWorkflow
+from openassessment.workflow.serializers import TeamAssessmentWorkflowSerializer
+
+
+logger = logging.getLogger('openassessment.workflow.models')  # pylint: disable=invalid-name
+
+
+def create_workflow(team_submission_uuid):
+    """
+    A team submission should only be assessible by staff.  Therefore, unlike
+    the analogous `create_workflow()` method for individual submissions,
+    we don't accept `steps` or `on_init_params` as parameters to this function,
+    since those are only used to indicate which assessment steps (e.g. "peer", "self")
+    are to be included in the workflow.
+    """
+    try:
+        team_workflow = TeamAssessmentWorkflow.start_workflow(team_submission_uuid)
+        logger.info((
+            u"Started team assessment workflow for "
+            u"team submission UUID {uuid}"
+        ).format(uuid=team_submission_uuid))
+        return team_workflow
+    except Exception:
+        err_msg = (
+            u"An unexpected error occurred while creating "
+            u"the workflow for team submission UUID {uuid}"
+        ).format(uuid=team_submission_uuid)
+        logger.exception(err_msg)
+        raise AssessmentWorkflowInternalError(err_msg)
+
+
+def get_workflow_for_submission(team_submission_uuid):
+    """
+    Like `api.get_workflow_for_submission`, but for teams.  We don't need
+    an analogous `assessment_requirements` parameter, because team submissions
+    are only assessible by staff (where requirements like "must_grade" and
+    "must_be_graded_by" are not supported).
+    """
+    # Get the wokflow for this submission
+    team_workflow = _get_workflow_model(team_submission_uuid)
+
+    # Update the workflow status based on the underlying assessments
+    try:
+        team_workflow.update_from_assessments()
+        logger.info((
+            u"Updated workflow for team submission UUID {uuid} "
+        ).format(uuid=team_submission_uuid))
+    except Exception as exc:
+        err_msg = (
+            u"Could not update team assessment workflow: {exc}"
+        ).format(exc=exc)
+        logger.exception(err_msg)
+        raise AssessmentWorkflowInternalError(err_msg)
```

Comment:
Would it make sense to have TeamAssessmentWorfklowInternalError, so we can differentiate between vanilla assessment and team specific issues?

Reply:
I've intentionally left these kind of vague since I don't know what the model level will raise or if we want to rethink any error-handling. That said, I feel like creating a new error class is overkill.

Reply:
Sounds good.